### ISSUE TITLE
release: v2.1.10

### ISSUE THREAD

**Issue by dt** [cockroachdb/cockroach#43063]
Candidate SHA: 9ad9eb5fb8806e4b74546910ca8bda66786d4288
Deployment status: [v2.1.9](http://david-release-v219-42d307e1-0001.roachprod.crdb.io:26258/#/overview/list) vs [v2.1.10](http://david-release-v2110-9ad9eb5f-0001.roachprod.crdb.io:26258/#/overview/list)
Nightly Suite: https://teamcity.cockroachdb.com/viewLog.html?buildId=1634847&buildTypeId=Cockroach_Nightlies_NightlySuite
 
[Release process checklist](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process)
 
Prep date: `2019-12-09`
- [x] [Pick a SHA](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-PickingaSHA), fill in `Candidate SHA` above, notify #release-announce of SHA.
- [x] [Tag the provisional SHA](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-TagtheprovisionalSHA)
- [x] [Publish provisional binaries](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-Publishtheprovisionalbinaries)
- [x] [Check binaries](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-Checkbinaries)
- [x] [Deploy to test clusters](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-DeploytoTestClusters)
- [ ] [Verify node crash reports](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-Verifynodecrashreportsappearinsentry.io)
- [x] [Start release qualification suite](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-RuntheSuiteofNightlyRoachtests)
- [x] [Start nightly suite](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-RuntheSuiteofNightlyRoachtests)
- [x] fill in `Deployment status` above with clusters and `Nightly Suite` with the link to Nightly TeamCity Job
 
One day after prep date:
- [x] [Get signoff on roachtest failures](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-RoachtestFailureSignoff)
 
- Keep an eye on clusters until release date. Do not proceed below until the release date.
 
Release date: `WIP fill in date: usually Monday WIP`
- [x] [Check cluster status](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-Checkclusterstatus)
- [x] [Tag release](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-Tagtherelease)
- [x] [Bless provisional binaries](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-Blesstheprovisionalbinaries)
 
- For production or stable releases in the latest major release series
  - [ ] Update the [Homebrew Formula](https://github.com/cockroachdb/homebrew-tap/blob/master/Formula/cockroach.rb)
  - [ ] Update the image tag in our [Orchestrator configurations](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73105625/Release+process#Releaseprocess-Orchestratorconfigurations)
  - [ ] Announce version to the [registration cluster](https://github.com/cockroachlabs/production#publishing-a-new-release-version)
  - [ ] Upgrade the registration cluster (reg runs latest stable version)
 
- [x] Update docs
- [x] External communications for release

**Comment by dt** [comment]
[ ] [import/tpcc/warehouses=1000/nodes=32](https://teamcity.cockroachdb.com/viewLog.html?buildId=1634847&buildTypeId=Cockroach_Nightlies_NightlySuite#testNameId-4551784729715115627 "Show related test details") 
[ ] [scaledata/filesystem_simulator/nodes=6](https://teamcity.cockroachdb.com/viewLog.html?buildId=1634847&buildTypeId=Cockroach_Nightlies_NightlySuite#testNameId1018105635407851240 "Show related test details")
[ ] [hibernate](https://teamcity.cockroachdb.com/viewLog.html?buildId=1634847&buildTypeId=Cockroach_Nightlies_NightlySuite#testNameId-3931757827174594180 "Show related test details")
[ ] [pgjdbc](https://teamcity.cockroachdb.com/viewLog.html?buildId=1634847&buildTypeId=Cockroach_Nightlies_NightlySuite#testNameId-5749868698903386485 "Show related test details")
[ ] [tpccbench/nodes=9/cpu=4/chaos/partition](https://teamcity.cockroachdb.com/viewLog.html?buildId=1634847&buildTypeId=Cockroach_Nightlies_NightlySuite#testNameId7647859149799979980 "Show related test details")

**Comment by dt** [comment]
@nvanbenschoten do you own the scaladata Roachtests or is there a team/slack channel I should ping to triage that timeout?

**Comment by nvanbenschoten** [comment]
@dt I took a look at the scaledata failure and it looks more like a testing flake than an actual issue, but I'm not 100% sure.

The test is cycling through nodes and restarting them one-by-one. We see that the cluster ends up (10 min into the test) in a healthy state.
```
4: 4060
1: 4284
5: 4343
6: 3881
3: 3552
2: 4085
5: kill exited nonzero
5: dead
5: 4835
2: kill exited nonzero
2: dead
2: 4702
4: kill exited nonzero
4: dead
4: 4812
2: kill exited nonzero
2: dead
2: 5249
4: kill exited nonzero
4: dead
4: 5321
```

It then appears to get stuck and sit there for 10 minutes until the test times out:
```
09:34:55 cluster.go:328: > /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod start teamcity-1634845-1575958471-63-n7cpu4:4
teamcity-1634845-1575958471-63-n7cpu4: starting.
CHAOS: 09:34:57 chaos.go:62: chaos stopping: <nil>
4: 5321
09:44:29 test.go:284: test failure: 	test_runner.go:715: test timed out (20m0s)
```

I'm going to try to reproduce. `scaledata/filesystem_simulator/nodes=3` passed, so whatever it appears to be flaky at best.

**Comment by dt** [comment]
@nvanbenschoten Thanks for digging into it!

**Comment by nvanbenschoten** [comment]
I was able to reproduce. I suspect that this is some kind of deadlock in the shutdown code of `scaledata/filesystem_simulator`.

**Comment by nvanbenschoten** [comment]
Although, interestingly, the only times that we've seen this failure mode in the past have been on the `release-2.1` branch. We've seen it 6 times in the past few months.

I'm also not able to reproduce when I drop the test time and chaos times down to try to get faster repros.

**Comment by nvanbenschoten** [comment]
I'm not getting anywhere with this. It doesn't look to be a new issue with 2.1 and it also seems contained to 2.1, so I'm not going to dig in any further. I think it's safe to sign off on for this release.

**Comment by jseldess** [comment]
@dt, can this be closed?
